lib/iris/cube.py

```diff
@@ -2263,8 +2181,18 @@ def new_cell_measure_dims(cm_):
             if ma.count_masked(data) == 0:
                 data = data.filled()
 
+        # XXX: Slicing a single item from a masked array that is masked,
+        #      results in numpy (v1.11.1) *always* returning a MaskedConstant
+        #      with a dtype of float64, regardless of the original masked
+        #      array dtype!
+        if isinstance(data, ma.core.MaskedConstant) and \
+                data.dtype != cube_data.dtype:
+            data = ma.array(data.data, mask=data.mask, dtype=cube_data.dtype)
+
```

Comment:
@dkillick We're pretty open to this behaviour anywhere, but we certainly really care about it here.

I do have a note in the [DataManager Integration](https://github.com/SciTools/iris/projects/5) project, that hints that we may want to add a `__getitem__` to the `DataManager` for convenience.

The evidence for this well come when we integrate the `DataManager` into coordinates and cell-measures ... but if we do then, this nasty case is good evidence to centralize the fix within the `DataManager.__getitem__`.

... and no, I've not search the whole code base looking for where else this might sting us. The thought depresses me :cry: